### PR TITLE
infra/config: Move tasks to the beginning of each lab

### PR DESCRIPTION
**Pull request by teodutu**
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/cs-pub-ro/operating-systems/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

Move the "Tasks" section at the start of each one-page lab so students aren't tempted to waste time reading the documentation during practical sessions when they should be going straight to the tasks.

### CODE REVIEW

**Comment by teodutu** [comment]
The lab generator script is hardcoded to have tasks at the end [1]. I'll need to amend it to accommodate variable section orders.

[1] https://github.com/cs-pub-ro/operating-systems/blob/13959d127634d81335a2462e6db71b834773b5a7/gen-view.py#L158-L171

**Comment by Alex-deVis** [comment]
> I'll need to amend it to accommodate variable section orders.

Just go ahead and hardcode tasks first, no reason to make the script 'smarter' right now.
Once we know how much customization is required we'll change the script.

File: config.yaml [hunk 1]
```diff
@@ -13,32 +13,29 @@ output_dir: /output
...
       - basic-syscall.md
       - syscall-wrapper.md
       - libcall-syscall.md
-    tasks:
+    content:
+      - modern-sw-stack.md
       - basic-syscall.md
       - syscall-wrapper.md
       - libcall-syscall.md
   - title: Lab 2 - Library Perspective
```
Comment (Alex-deVis) [inline config.yaml:25]: While you are doing this, consider adding the long lost guides for this lab. @mariapana would appreciate.

**Comment by Alex-deVis** [comment]
Just merged #116. Rebase and reorder sections to get this done.

File: config.yaml [hunk 2]
```diff
@@ -13,39 +13,50 @@ output_dir: /output
 lab_structure:
   - title: Lab 1 - Operating System Perspective
     filename: lab1.md
-    content:
-      - modern-sw-stack.md
+    tasks:
       - basic-syscall.md
       - syscall-wrapper.md
       - libcall-syscall.md
```
Comment (Alex-deVis) [inline config.yaml:19]: Remove these.

File: config.yaml [hunk 3]
```diff
@@ -13,39 +13,50 @@ output_dir: /output
...
     content:
-      - common-functions.md
-      - libc.md
-      - high-level-lang.md
-      - app-investigate.md
       - tasks/common-functions.md
       - tasks/libc.md
       - tasks/high-level-lang.md
       - tasks/app-investigation.md
+      - common-functions.md
```
Comment (Alex-deVis) [inline config.yaml:35]: This was deleted by mistake when `gen-view.py` was added.
```suggestion
      - guides/static-dynamic.md
      - common-functions.md
```

**Review by Alex-deVis — CHANGES_REQUESTED** [review]

**Review by Alex-deVis — APPROVED** [review]

**Pull request merged**